### PR TITLE
upgrade for new survey machenism

### PR DESCRIPTION
**Pull request by Nickcandy**
## Description
1. Add InstallationId for intercept survey, kept in AzureRmContextSettings.json.
2. When importing Az module, maintain relative information about intercept survey.
3. Change the redirect URL for Open-AzSurveyLink. 

Append fix of #18104 in common lib

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Accounts/Authentication/AzureSessionInitializer.cs [hunk 1]
```diff
@@ -202,6 +212,21 @@ static ContextAutosaveSettings InitializeSessionSettings(IDataStore store, strin
             return result;
         }
 
+        static String InitializeInstallationId(){
```
Comment (dingmeng-xue) [inline src/Accounts/Authentication/AzureSessionInitializer.cs:215]:
```suggestion
        static String GetAzureCLIInstallationId(){
```

File: src/Accounts/Authentication/AzureSessionInitializer.cs [hunk 2]
```diff
@@ -173,7 +175,13 @@ static ContextAutosaveSettings InitializeSessionSettings(IDataStore store, strin
                     result.ContextDirectory = migrated ? profileDirectory : settings.ContextDirectory ?? result.ContextDirectory;
                     result.Mode = settings.Mode;
                     result.ContextFile = settings.ContextFile ?? result.ContextFile;
-                    if (migrated)
+                    result.Settings = settings.Settings;
+                    String installationId = settings.Settings.ContainsKey("InstallationId") ?settings.Settings["InstallationId"] : null;
```
Comment (dingmeng-xue) [inline src/Accounts/Authentication/AzureSessionInitializer.cs:179]:
```suggestion
                    String installationId = settings.Settings.ContainsKey("InstallationId") ?settings.Settings["InstallationId"] : GetAzureCLIInstallationId();
                    installationId = installationId ?? Guid.NewGuid().ToString();
```

File: src/Accounts/Accounts/Feedback/OpenAzSurveyLinkCmdlet.cs [hunk 3]
```diff
@@ -13,28 +13,58 @@
...
 
         protected override string DataCollectionWarning => null;
 
         public override void ExecuteCmdlet()
-        {
-            WriteInformation(new HostInformationMessage() { Message = $"Opening the default browser to {_surveyLinkFormat}" }, new string[] { "PSHOST" });
-            OpenBrowser(_surveyLinkFormat);
+        {   
+            StreamReader sr = null;
+            DateTime Today = DateTime.UtcNow;
```
Comment (isra-fel) [inline src/Accounts/Accounts/Feedback/OpenAzSurveyLinkCmdlet.cs:43]: Use lowercase for local variables
```suggestion
            DateTime today = DateTime.UtcNow;
```

File: src/Accounts/Authentication/AzProfileInfo.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Common.Authentication
+{
+    internal class AzProfileInfo
```
Comment (isra-fel) [inline src/Accounts/Authentication/AzProfileInfo.cs:20]: Move this file into './Models'

File: src/Accounts/Authentication/AzureSessionInitializer.cs [hunk 5]
```diff
@@ -39,6 +39,8 @@ namespace Microsoft.Azure.Commands.Common.Authentication
     public static class AzureSessionInitializer
     {
         private const string ContextAutosaveSettingFileName = ContextAutosaveSettings.AutoSaveSettingsFile;
+
+        private static string AzProfileInfoFile = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),".Azure", "AzureProfile.json");
```
Comment (isra-fel) [inline src/Accounts/Authentication/AzureSessionInitializer.cs:43]:
```suggestion
        private static readonly string AzProfileInfoFile = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),".Azure", "AzureProfile.json");
```

File: src/Accounts/Authentication/AzureSessionInitializer.cs [hunk 6]
```diff
@@ -202,6 +211,18 @@ static ContextAutosaveSettings InitializeSessionSettings(IDataStore store, strin
             return result;
         }
 
+        static String GetAzureCLIInstallationId(){
+            String installationId = null;
+            if (File.Exists(AzProfileInfoFile))
```
Comment (isra-fel) [inline src/Accounts/Authentication/AzureSessionInitializer.cs:216]: Always use the IDataStore abstraction
```suggestion
        static String GetAzureCLIInstallationId(IDataStore store) {
            String installationId = null;
            if (store.FileExists(AzProfileInfoFile))
```

**Review by isra-fel — CHANGES_REQUESTED** [review]
I suggest using the format code function of your IDE whenever you update a file.

File: src/Accounts/Accounts/Feedback/ScheduleInfo.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
+﻿// ----------------------------------------------------------------------------------
```
Comment (dingmeng-xue) [inline src/Accounts/Accounts/Feedback/ScheduleInfo.cs:1]: Do you need this file? Class in Share lib can be reused here.

File: src/Accounts/Authentication/Config/Definitions/EnableInterceptSurveyConfig.cs [hunk 8]
```diff
@@ -36,14 +36,10 @@ internal class EnableInterceptSurveyConfig : TypedConfig<bool>
```
Comment (dingmeng-xue) [inline src/Accounts/Authentication/Config/Definitions/EnableInterceptSurveyConfig.cs:36]: Please rename this class name.

File: src/Accounts/Authentication/Models/AzProfileInfo.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Models
+{
+    internal class AzProfileInfo
```
Comment (dingmeng-xue) [inline src/Accounts/Authentication/Models/AzProfileInfo.cs:20]: => AzureCLIProfileInfo

File: ChangeLog.md [hunk 10]
```diff
@@ -2,6 +2,8 @@
 #### Az.Accounts
 * Implemented 'SupportsShouldProcess' for 'Invoke-AzRestMethod'
 * Supported giving suggestions if an Azure PowerShell command cannot be found, for example when there is a typo in command name.
+* Updated survey message and settings
```
Comment (dingmeng-xue) [inline ChangeLog.md:5]: wrong location.

**Review by dingmeng-xue — APPROVED** [review]

**Pull request merged**